### PR TITLE
feat:add hr-profile table [WIP]

### PR DESCRIPTION
**Pull request by casterweb**
image

### CODE REVIEW

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts [hunk 1]
```diff
@@ -1,10 +1,93 @@
...
+    const users = Array.from({ length: 100 }, (_, k) => createNewUser(k + 1));
+
+    // Assign the data to the data source for the table to render
+    this.dataSource = new MatTableDataSource(users);
+  }
+
+  ngAfterViewInit() {
+    // @ts-ignore
+    this.dataSource.paginator = this.paginator;
+    // @ts-ignore
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:70]: Its not the best idea to ignore errors
Could you find out how to fix it?

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts [hunk 2]
```diff
@@ -1,10 +1,93 @@
...
+    const filterValue = (event.target as HTMLInputElement).value;
+    this.dataSource.filter = filterValue.trim().toLowerCase();
+
+    if (this.dataSource.paginator) {
+      this.dataSource.paginator.firstPage();
+    }
+  }
+}
+
+/** Builds and returns a new User. */
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:84]: Its not the good practice to have comments for function, function name should be fully descriptive itself

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts [hunk 3]
```diff
@@ -1,10 +1,93 @@
...
+import { MatTableDataSource } from '@angular/material/table';
+import { MatSort } from '@angular/material/sort';
+
+export interface UserData {
+  id: string;
+  name: string;
+  lastName: string;
+}
+
+const LastName: string[] = [
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:12]: Why naming style is different, lets make them MOCK_NAMES and MOCK_LAST_NAMES

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.scss [hunk 4]
```diff
@@ -0,0 +1,16 @@
+table {
+  width: 100%;
+}
+.mat-elevation-z8{
+margin: 0 30px;
+}
+
+.mat-form-field {
+  background-color: #424242;
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.scss:9]: We have $secondary-grey1 variable for that color

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts (hunk 2 above)
Comment (casterweb) [reply exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:84]: image
done

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts (hunk 1 above)
Comment (casterweb) [reply exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:70]: image
done

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.scss (hunk 4 above)
Comment (casterweb) [reply exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.scss:9]: image
done

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts (hunk 3 above)
Comment (casterweb) [reply exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:12]: image
done

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts [hunk 5]
```diff
@@ -1,10 +1,91 @@
...
+  dataSource: MatTableDataSource<UserData>;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator | null = null;
+
+  @ViewChild(MatSort) sort: MatSort | null = null;
+
+  constructor() {
+    // Create 100 users
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    const users = Array.from({ length: 100 }, (_, k) => createNewUser(k + 1));
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:61]: Just put createNewUser definition on top of file (even better to some src/mocks/users-utils.mock.ts)

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts [hunk 6]
```diff
@@ -1,10 +1,91 @@
...
+
+  dataSource: MatTableDataSource<UserData>;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator | null = null;
+
+  @ViewChild(MatSort) sort: MatSort | null = null;
+
+  constructor() {
+    // Create 100 users
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:60]: Just put createNewUser definition on top of file (even better to some src/mocks/users-utils.mock.ts)

File: exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts [hunk 7]
```diff
@@ -1,10 +1,91 @@
...
+    const users = Array.from({ length: 100 }, (_, k) => createNewUser(k + 1));
+
+    // Assign the data to the data source for the table to render
```
Comment (RomPavel) [inline exadel-trainee-app/src/app/models/hr-profile/hr-profile.component.ts:63]: Having comments is antipattern, the code itself should be easy readable and understandable.
If you think that without comment others will not get idea of your code, than itss time to rename variables, functions and so on

**Review by Inunor — APPROVED** [review]

**Pull request merged**